akka-docs/src/main/paradox/typed/cluster-membership.md

```diff
@@ -126,6 +130,27 @@ that are in this state, but you should be aware of that members on the other
 side of a network partition have no knowledge about the existence of the
 new members. You should for example not count `WeaklyUp` members in quorum decisions.
 
+## Full cluster shutdown
+
+In some rare cases it may be desirable to do a full cluster shutdown rather than a rolling deploy. 
+For example, a protocol change where it is simpler to restart the cluster than to make the protocol change
+backward compatible.
+
+As of Akka `2.6.12` it can be signalled that a full cluster shutdown is about to happen and any expensive actions such as:
+
+* Cluster sharding rebalances
+* Moving of Cluster singletons
+
+Won't happen. That way the shutdown will be as quick as possible and a new version can be started up without delay.
+
+If a cluster isn't to be restarted right away then there is no need to prepare it for shutdown.
+
+To use this feature use `Cluster(system).prepareForFullClusterShutdown()` in classic or @apidoc[PrepareForFullClusterShutdown] in typed.
+
+Wait for all `Up` members to become `ReadyForShutdown` and then all nodes can be shutdown and restarted.
+Members that aren't `Up` yet will remain in the `Joining` or `WeaklyUp` states. Any node that is already leaving 
+the cluster i.e. in the `Leaving` or `Exiting` states will continue to leave the cluster via the normal path.
```

Comment:
what is the expected flow for the users? What should they do when all members become `ReadyForShutdown`? Should they scale to 0 at that point? 

If we are dependent of users checking the status of all nodes and then acting to scale to 0, it will probably take longer then simply scale directly to 0 and accept with the overhead, no?

Reply:
The purpose of the feature is to make the time between shutting down and starting up as small as possible. This time starts when the user shuts down the nodes, not when the user prepares for shutdown

Reply:
We can't shutdown as in most environments if we shutdown they'll be restarted